### PR TITLE
Add `not-windows-x64` category to `archos/`

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr adds the `nonwindows-x64` category to `archos/`. This is to:
1. Prevent duplicate tests in `linux-x64` and `darwin-x64`.
2. Allow the BSD OSes to join in on the Linux/Darwin testing party as well.

Imho it's quite obvious that mainstream OSes can generally be classified into 2 categories: Windows and Unix-like. Imo if anything more complicated than this is needed (e.g. not Windows and not OpenBSD), it should be done as part of a unit test.

**Test plan**

All builds are green. Tests in `archos/nonwindows-x64` are run as appropriate.

**Closing issues**

...

### CODE REVIEW

**Comment by wargio** [comment]
how about just defining `not-` prefix ?

**Comment by kazarmy** [comment]
the negation only applies to the `windows` part, not the `x64` part

**Comment by wargio** [comment]
yes, and this is my point, so we can negate any arch

**Comment by kazarmy** [comment]
`not-windows-x64` does also include `windows-x86`, which is not the meaning here

**Comment by wargio** [comment]
hmmmm i understand what you are referring to, but we can just limit that not to arch and not bits

**Comment by kazarmy** [comment]
ok as you wish 9e9e1a4711800d8fb4263961cd50d33b13b94623 ... btw, `windows` is the os, and `x64` is arch and bitness combined

**Comment by wargio** [comment]
actually what i meant is that if the filename starts with `not-` then we negate the OS check.
example:
- `not-linux-x86` means run on any OS that is **not linux** which has arch `x86`
- `not-darwin-arm` means run on an OS that is **not darwin** which has arch `arm`

**Comment by kazarmy** [comment]
that's what i understood too, and `not-windows-x64` is exactly according to what you meant

File: binrz/rz-test/load.c [hunk 1]
```diff
@@ -612,7 +612,10 @@ static bool database_load(RzTestDatabase *db, const char *path, int depth) {
 				eprintf("Skipping %s" RZ_SYS_DIR "%s because it requires additional dependencies.\n", path, subname);
 				continue;
 			}
-			if ((!strcmp(path, "archos") || rz_str_endswith(path, RZ_SYS_DIR "archos")) && strcmp(subname, RZ_TEST_ARCH_OS)) {
+			if ((!strcmp(path, "archos") || rz_str_endswith(path, RZ_SYS_DIR "archos")) && strcmp(subname, RZ_TEST_ARCH_OS) &&
+				!(!strcmp(subname, "not-windows-x64") &&
+					!rz_str_startswith(RZ_TEST_ARCH_OS, "windows-") &&
+					rz_str_endswith(RZ_TEST_ARCH_OS, "-x64"))) {
```
Comment (ret2libc) [inline binrz/rz-test/load.c]: do we really need the specific reference to windows here? Can't we rewrite it with something like `startswith(not-) && strstr(RZ_TEST_ARCH_OS, second-part-of-the-path)`?

Comment (kazarmy) [reply binrz/rz-test/load.c]: no, yes we can 3794dbd2d457c240297aff94468b3e85fa7d5f89

**Review by wargio — APPROVED** [review]

**Pull request merged**